tests/capture/workflow/test_capture_qnode.py

```diff
@@ -1218,3 +1217,437 @@ def circuit():
             return qml.state()
 
         assert qml.math.allclose(circuit(), [0, 0, 0, 1])
+
+
+def test_qnode_static_argnums():
```

Comment:
Can we add a test that would have failed before you added the `sort` to `qnode.static_argnums`?